File: backend/bodhlens-ai/src/main/java/com/bodhlens/bodhlens_ai/service/ChatService.java
package com.bodhlens.bodhlens_ai.service;

import java.io.IOException;
import java.util.UUID;

import com.bodhlens.bodhlens_ai.dto.ChatRequest;
import com.bodhlens.bodhlens_ai.dto.ChatResponse;

public interface ChatService {

	ChatResponse chat(UUID documentId, ChatRequest request) throws IOException, InterruptedException;
}

File: frontend/bodhlens-ui/src/app/services/chat-service.ts
import { HttpClient } from '@angular/common/http';
import { inject, Service } from '@angular/core';
import { ChatRequest } from '../models/chat-request';
import { Observable } from 'rxjs';
import { ChatResponse } from '../models/chat-response';

@Service()
export class ChatService {

    private httpClient = inject(HttpClient);

    private apiUrl = `http://localhost:8080/api/chat`; 

    chat(documentId: string, req: ChatRequest): Observable<ChatResponse> {
        return this.httpClient.post<ChatResponse>(
            `${this.apiUrl}/${documentId}`,
            req
        );
    }
}
